records/2023-11-21.quicksight.md

```diff
@@ -0,0 +1,56 @@
+# QuickSight
+
+Date: 2023-11-21
+
+## Status
+
+**DRAFT**.
+
+## Context
+
+To have a thorough understanding of data in our system, the Notify team wanted a collaborative tool that would allow them to explore the database. While we already have installed Blazer, we need a tool with more business intelligence capabilities that is more accessible to non-developers. AWS QuickSight was chosen to be our short/mid term solution for a BI tool that has no sensitive data, that external CDS teams (ie other than Notify) can tap into, and which is easily queryable and fast.
+
+## Implementation
+
+Introducing QuickSight into our system requires us to do the following:
+
+1. QuickSight needs access to the main Notify DB
+2. QuickSight needs to be secure and auditable
+
+## Deployment
+
+### QuickSight
+
+QuickSight is an AWS product that we can run in ca-central-1. We can use terraform to create and modify AWS Quicksight subscriptions in our accounts.
+
+### Data
+
+QuickSight imports data from the Notify database into optimized data storage called SPICE. This allows queries to run faster and without creating a load on RDS. SPICE datasets (ie tables) are updated either manually or on schedule. Updates can either be full or incremental.
+
+We specify the data fields added to the QuickSight datasets. We are excluding any fields containing PII or user data (for example, email addresses, phone numbers, and template personalisation).
```

Comment:
While I understand why we would do this (and I'm not really suggesting we shouldn't), do you think it will hamper our analysis efforts at all?  Will anyone outside of CDS have access to quicksight?

Reply:
Employees of CDS should not have access to the protected data of the users of our system, I think we would have to have very strong justifications and access controls in place before we considered including this data in quicksight.

Reply:
I generally agree, I guess why I'm asking is because the "to" email address is considered PII but we can probably get a lot of insights from it.  Maybe we could just show the domain rather than the full email address or something? Know what I mean?

Reply:
Yeah there are some queries that are quite useful around domains as you mention @andrewleith . We can perform transformations from the DB to Quicksight, hence we'll finetune things as we learn. 

It's an area too where a OLAP schema with fact tables would be best, as we're losing some valuable information when we archive the emails. We could transform the data into domains to remove PIIs and dump summaries in there. We could do it now inside our current schema as this is where we host fact tables at the moment.

Reply:
For the V1 of QuickSight I think we should restrict to no PII, using Blazer for any PII related queries.